test/ValidateID.cpp

```diff
@@ -1344,19 +1345,71 @@ TEST_F(ValidateID, OpReturnValueVariableGood) {
      OpFunctionEnd)";
   CHECK(spirv, SPV_SUCCESS);
```

Comment:
This example looks wrong. Validation should fail. ?
  %7 is the result of an OpVariable, and hence is a pointer to 32-bit int.
But the function returns 32-bit int (%2), and the return value is a pointer-to-32-bit int in Function storage class.